src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.UnifiedBuild.Tasks/ManifestAssets/JoinVerticalsAssetSelector.cs

```diff
@@ -85,8 +77,8 @@ bool IsExternalAsset(ManifestAsset asset)
                         AssetId = assetGroup.Key,
                         MatchType = matchType,
                         VerticalName = verticalName,
-                        Asset = assetGroup.FirstOrDefault(o => VerticalNameMatches(o.manifest.VerticalName, verticalName)).asset,
-                        OtherVerticals = assetGroup.Select(o => o.manifest.VerticalName!).Where(o => !VerticalNameMatches(o, verticalName)).ToList()
+                        Asset = assetGroup.FirstOrDefault().asset,
```

Comment:
The enclosing if...when would the verticalCount be 0?

Reply:
Good point, that shouldn't ever happen.

I'll clean this up.